tagcompare/capture.py

```
@@ -14,11 +17,164 @@
+            return None
+
+        tag_html = tags_per_campaign[pathbuilder.tagsize][pathbuilder.tagtype]
+        pathbuilder.create()
+        output_path = pathbuilder.tagimage
+        errors = self.capture_tag(tag_html=tag_html, output_path=output_path,
+                                  tagtype=pathbuilder.tagtype)
+        self.__write_html(
+            tag_html=tag_html, output_path=pathbuilder.taghtml)
+        return errors
+
+    def capture_tags(self, tags, pathbuilder,
+                     tagsizes=settings.DEFAULT.tagsizes,
+                     tagtypes=settings.DEFAULT.tagtypes,
+                     capture_existing=False):
+        num_existing_skipped = 0
+        num_captured = 0
+        browser_errors = []
+
+        try:
+            for cid in tags:
+                pathbuilder.cid = cid
+                tags_per_campaign = tags[cid]
+                # LOGGER.debug("tags_per_campaign: %s", str(tags_per_campaign))
+                # TODO: Refactor better with _capture_tag
+                # It's weird that we pass in a pathbuilder object and do two
+                # nested loops here
+                for tagsize in tagsizes:
+                    if tagsize not in tags_per_campaign:
+                        LOGGER.warn("No tagsize '%s' found for campaign: %s. Skipping",
+                                    tagsize, cid)
+                        continue
+                    pathbuilder.tagsize = tagsize
+                    for tagtype in tagtypes:
+                        pathbuilder.tagtype = tagtype
+                        r = self._capture_tag(pathbuilder=pathbuilder,
+                                              tags_per_campaign=tags_per_campaign,
+                                              capture_existing=capture_existing)
+                        if r is None:
+                            num_existing_skipped += 1
+                        elif r is False:
+                            continue
```

I was thinking you'd not worry about handling the exception in `_capture_tag()`, because it doesn't really know how to handle them. Let them propagate here, and handle them here, because here you know what to do with them:

  - log the failure
  - increment the error counter
  - move on to the next thing